src/components/shared/input/index.tsx

```diff
@@ -50,12 +56,23 @@ const Input = <
             }}
             placeholder={placeholder}
             autoCapitalize={autocapitalize}
-            type={type}
+            type={type === 'password' && isPasswordVisible ? 'text' : type}
```

Comment:
A useMemo might be a better approach to add this piece of logic, making the component logic more clear